packages/dev/test-utils/src/dnd.ts

```diff
@@ -0,0 +1,114 @@
+import {act, fireEvent} from '@testing-library/react';
+// eslint-disable-next-line monorepo/no-internal-import
+import {DataTransfer, DragEvent} from '@react-aria/dnd/test/mocks';
+
+type DragPointerType = 'mouse' | 'touch' | 'pointer';
+
+type Delta = {x: number, y: number}
+
+function getElementCenter(element: Element) {
+  let {left, top, width, height} = element.getBoundingClientRect();
+  return {
+    x: left + width / 2,
+    y: top + height / 2
+  };
+}
+
+async function performDrag(source: Element, target: Element, delta: Delta, type: DragPointerType = 'mouse', steps = 1, duration = 0) {
+  let from = getElementCenter(source);
+  let to = {x: 0, y: 0};
+
+  if (target) {
+    to = getElementCenter(target);
+  } else if (delta) {
+    to = {x: from.x + delta.x, y: from.y + delta.y};
+  }
+
+
+  let step = {
+    x: (to.x - from.x) / steps,
+    y: (to.y - from.y) / steps
+  };
+
+
+  let current = {
+    clientX: from.x,
+    clientY: from.y
+  };
+
+  if (type === 'mouse') {
+    fireEvent.mouseEnter(source, current);
+    fireEvent.mouseOver(source, current);
+    fireEvent.mouseMove(source, current);
+    fireEvent.mouseDown(source, current);
+  } else if (type === 'pointer') {
+    fireEvent.pointerEnter(source, current);
+    fireEvent.pointerOver(source, current);
+    fireEvent.pointerMove(source, current);
+    fireEvent.pointerDown(source, current);
+  } else if (type === 'touch') {
+    fireEvent.touchStart(source, current);
+    fireEvent.touchMove(source, current);
+  }
+
+  let dataTransfer = new DataTransfer();
+  fireEvent(source, new DragEvent('dragstart', {dataTransfer, clientX: current.clientX, clientY: current.clientY}));
+  await act(async () => Promise.resolve());
+
+  for (let i = 0; i < steps; i++) {
+    current.clientX += step.x;
+    current.clientY += step.y;
+
+    if (duration !== 0 && steps > 1) {
+      await new Promise(resolve => {
+        setTimeout(resolve, duration / steps);
```

Comment:
this is going to be reliant on people either mocking the clock or not mocking it. For instance, we mock the clock, but I don't see a place for us to advance it here